### PR TITLE
Vee two

### PR DESCRIPTION
**Pull request by pushplay**
This is currently a work in progress but comments are welcome.

### CODE REVIEW

**Comment by Tim-Jordison** [comment]
Nice work on these changes Jeff. Everything reads well and I like the updates to NetworkProvider. Less code is a bonus. Also, simple JUnit tests seem appropriate for this project.

File: README.md [hunk 1]
```diff
@@ -1,315 +1,79 @@
 # Lightrail Client for Java
 
-Lightrail is a modern platform for digital account credits, gift cards, promotions, and points (to learn more, visit Lightrail). This is a basic library for developers to easily connect with the Lightrail API using Java. If you are looking for specific use cases or other languages, check out the complete list of all Lightrail [libraries and integrations](https://www.lightrail.com/docs/#client-libraries/client-libraries).
-
+Lightrail is a modern platform for digital account credits, gift cards, promotions, and points (to learn more, visit [Lightrail](https://www.lightrail.com/)). This is a basic library for developers to easily connect with the Lightrail API using Javascript or Typescript. If you are looking for specific use cases or other languages, check out the complete list of all [Lightrail libraries and integrations](https://github.com/Giftbit/Lightrail-API-Docs/blob/master/README.md#lightrail-integrations).
```
Comment (tjukes) [inline README.md]: Typo "Javascript or Typescript" should be "Java"

Comment (tjukes) [reply README.md]: Should probably mention checkout. 
Could just replace the first sentence with the first sentence from the docs: "Lightrail makes it easy to run promotions, manage customer loyalty points, create gift cards, and collect payment all in one unified checkout solution (to learn more, visit [Lightrail](https://www.lightrail.com/))."

Comment (tjukes) [reply README.md]: Also need to drop that last sentence with the broken integrations link.

File: README.md [hunk 2]
```diff
@@ -1,315 +1,79 @@
...
 
 ## Features
 
 The following features are supported in this version:
 
-- Account Credits: create, retrieve, balance-check, and create/simulate transactions.
-- Gift Cards: create, retrieve, balance-check, and create/simulate transactions.
+##### Contacts
+Create, Get, List, Update, Delete, List Values, Attach Value
```
Comment (tjukes) [inline README.md]: Looks like this is supposed to be `Values`. We don't have `delete`.

File: README.md [hunk 3]
```diff
@@ -1,315 +1,79 @@
...
+ 
+##### Values
+Create, Get, List, Update
```
Comment (tjukes) [inline README.md]: `Contacts`? We do have `delete` for Contacts. Also `attach`.

File: README.md [hunk 4]
```diff
@@ -1,315 +1,79 @@
...
 
-Note that the Lightrail API supports many other features and we are still working on covering them in this library. For a complete list of Lightrail API features check out the [Lightrail API documentation](https://www.lightrail.com/docs/). 
+##### Transactions
+Checkout, Debit, Credit, Transfer, Reverse, Capture Pending, Void Pending, Get, List
+  
+##### Currencies
+Create, Get, List, Update, Delete
 
+Note that the Lightrail API supports many other features and we are still working on covering them in this library. For a complete list of Lightrail API features check out the [Lightrail API documentation](https://www.lightrail.com/docs/).
```
Comment (tjukes) [inline README.md]: Link to https://lightrailapi.docs.apiary.io/#  ?

File: README.md [hunk 5]
```diff
@@ -1,315 +1,79 @@
...
 
 ## Installation
 
+This library requires Java 8 or higher.
```
Comment (tjukes) [inline README.md]: 👍

**Pull request merged**